### ISSUE TITLE
Which typeahead version?

### ISSUE THREAD

**Issue by oschonrock** [Nodws/bootstrap4-tagsinput#23]
Is bootstrap4-tagsinput supposed to work with typeahead. It seems to be "the best game in town" to get a tagsinput working wit bootstrap4 but I am struggling with typeahead. 

It "sort of works". I am using latest bootstrap4, plus latest bootstrap4-tagsinput plus typeahead 0.11.2 from npm, plus I am using this BS4 typeahead css to get close:

https://github.com/bassjobsen/typeahead.js-bootstrap4-css

But I am getting js errors in console:
"TypeError: dataset is null"
When selecting the typeahead drop down suggestions with the mouse (keyboard down arrow works much better). 

I think the typeahead.js API changed in this area at some point, so I wonder against which version of typeahead.js, bootstrap4-tagsinput is supposed to work? Or is this not tested?

Any information or help appreciated. Bit stuck after a couple of days on this.

**Comment by Nodws** [comment]
Try to change the Typeahead call on `examples/assets/app_bs3.js ` line 26  to
  ```
typeaheadjs: [
  {
    highlight: true,
    //other options
  },
  {
    name: 'citynames',
    displayKey: 'name',
    valueKey: 'name',
    source: citynames.ttAdapter()
  }]
```

**Comment by oschonrock** [comment]
Thanks for coming back.  I have got some progress!?
Initially I got: "uncaught exception: Can't add objects when itemValue option is not set"

so I added         

```js
itemValue: "name
```
 to tagsinput options. like so:

```js
    var availTags = new Bloodhound({
        datumTokenizer: Bloodhound.tokenizers.obj.whitespace('name'),
        queryTokenizer: Bloodhound.tokenizers.whitespace,
        prefetch: {
            url: '/rep/index.php?page=ajax&action=tag.tag_list',
            filter: function(list) {
                return $.map(list, function(availTag) {
                    return { name: availTag }; });
            }
        }
    });
    availTags.initialize();

    $('.avail-tags').tagsinput({
        typeaheadjs:[
            {
                highlight: true,
            },
            {
                name: 'availTags',
                displayKey: 'name',
                valueKey: 'name',
                source: availTags.ttAdapter(),
            }], 
        itemValue: true,

    })
```
That seems to make it "work". Thanks!

Now not getting any console errors and mouse click selecting from typeahead suggestions works. 

Next problems (unrelated you will no doubt say), are:

1. hitting Enter, submits the form with any setting of  cancelConfirmKeysOnEmpty
2. entering a value which is not in typeahead suggestions and hitting Enter submits form and new value is not in POST despite freeInput: true (the default). 

Any tips?

**Comment by Nodws** [comment]
Yes in fact that is a separate issue, but can be addressed by disabling the enter key
```
$('.avail-tags').on('keydown', function(e) {
    if (e.keyCode === 13) {
        return false
    }
});
```

or via the form's onSubmit event
